### PR TITLE
FIX/API Always set polymorphic class name in gridfield on new records.

### PR DESCRIPTION
**Pull request by GuySartorelli**
When creating a new record in a gridfield item edit form and polymorphic `has_one` relations have the ID set, but not the class. In polymorphic relations the ID is useless without the class.

This has caused at least one known issue (see parent issue below) where a relation is relied on for canX permissions but ultimately the relation only exists _after_ the record has been created.

## Parent issue
- silverstripe/silverstripe-userforms#1103

### CODE REVIEW

**Comment by michalkleiner** [comment]
I thought I submitted a PR for this in past, but could've been related to polymorphic has_one, not has_many.

Unrelated — I don't think we support two change types in the commit message such as FIX/API or ENH/DOC for automatic changelog creation, so you need to choose one or split the commit into two.

**Comment by GuySartorelli** [comment]
> I don't think we support two change types in the commit message

There are two commits, one for each the API and the FIX. I don't think the PR name itself will affect the changelog.
If I'm wrong about that then let me know and I'll change the changelog tag to `API` as the higher priority of the two.

**Comment by michalkleiner** [comment]
True, the issue name has no role in that, forgot to check the list of actual commits. All good.

**Review by sabina-talipova — APPROVED** [review]

File: src/ORM/PolymorphicHasManyList.php [hunk 1]
```diff
@@ -29,6 +29,14 @@ public function getForeignClass()
         return $this->dataQuery->getQueryParam('Foreign.Class');
     }
 
+    /**
+     * Gets the field name which holds the related object class.
+     */
+    public function getForeignClassKey(): string
```
Comment (maxime-rainville) [inline src/ORM/PolymorphicHasManyList.php:35]: Can we add an explicit test for this method.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Looks good ... I would like to see an explicit test to cover that new method.

@sabina-talipova Can you confirm that you've manually tested that this PR fixes the user form issue?

File: src/ORM/PolymorphicHasManyList.php (hunk 1 above)
Comment (GuySartorelli) [reply src/ORM/PolymorphicHasManyList.php:35]: Sure thing

**Comment by sabina-talipova** [comment]
I've done test in my local environment. Manual tests - PASSED (NO ERROR), phpUnit tests in local - PASSED.

**Comment by GuySartorelli** [comment]
@maxime-rainville I've added the test you requested.

File: src/ORM/PolymorphicHasManyList.php (hunk 1 above)
Comment (GuySartorelli) [reply src/ORM/PolymorphicHasManyList.php:35]: Test added.

**Pull request merged**